src/main/java/org/spongepowered/api/entity/ai/task/AITask.java

```diff
@@ -72,10 +73,10 @@
      * 3. Returning "true" will add the provided task to the list of updated tasks, if not there
      *    already.
      *
-     * @param other The other task
+     * @param that The other task
      * @return True if it can be, false if not
      */
-    boolean canRunConcurrentWith(AITask<O> other);
+    boolean canRunConcurrentWith(AITask<?> that);
```

Comment:
@ST-DDT 

```Well, what are relevant tasks? IMO only those how could be executed concurrently which this method will determine.```

Relevant being defined as a Task whose owner is, ideally, the same as this task's owner or, a bit worse, the same type as this task's owner.

If this was a "Vanilla only AI API", I'd not worry too much about some of this in the API and would ensure that any task you are passed would be the same owner as this task. Unfortunately, custom goals are an objective of the AI framework.

Strictly speaking, the compile errors are more of an implementation problem (as you said above) and not my problem as the API (not to be rude). This means ```? super O``` IS "better" than just ```?``` as you have an upper-bound. With just ```?```, any AITask with any type can be passed in.

With that said, I appreciate and get your sentiments (and am giving them serious thought). Even with the ability to do Custom Goals with the API, I may relax the reins a bit and let the plugin dev take heat if they start inter-twining tasks in such a way.